internal/icinga2/client.go

```diff
@@ -173,36 +224,57 @@ func (client *Client) buildHostServiceEvent(ctx context.Context, result CheckRes
 		}
 	}
 
-	ev, err := client.buildCommonEvent(ctx, host, service)
+	ev, err := client.buildCommonEvent(ctx, sc.Host, sc.Service)
 	if err != nil {
 		return nil, err
 	}
 
 	ev.Type = event.TypeState
 	ev.Severity = eventSeverity
-	ev.Message = result.Output
+	ev.Message = sc.CheckResult.Output
+
+	// Downtime suppresses all kind of events, so always generate a muted event if it's in downtime.
+	if sc.DowntimeDepth > 0 {
+		ev.SetMute(true, "Checkable is in downtime")
+	} else if sc.IsFlapping {
+		// If the Checkable is not in downtime but in flapping state, all other states/events become irrelevant as well.
+		ev.SetMute(true, "Checkable is in flapping state")
+	} else if sc.Acknowledgement {
+		// If the Checkable is acked, we've to always generate a muted event as well.
+		ev.SetMute(true, "Checkable problem state is acknowledged")
+	}
```

Comment:
Are you referring to just the `IsFlapping` part or all of them? Yes, currently `IsFlapping` is set for the catch-up phase only, but the others are not, and I want to have them centralised here as `buildHostServiceEvent()` is used for all state change events, so I don't quite get what you're asking for.

Reply:
I'm referring to all of them. These seem to be redundant with the handling of downtime start/flapping start/ack set events, so I was wondering if this is actually just for the catch-up phase where no special handling for these is done besides for acknowledgements.

If this was necessary (just) for the catch-up phase, I'd wonder about two more things: why is there no unmute happening otherwise and it looks like you might lose those mute events if no state change happened.

Reply:
> so I was wondering if this is actually just for the catch-up phase where no special handling for these is done besides for acknowledgements

As I said, this is not exclusive to the catch-up phase, but applies to all state change events and ensures that no notifications are triggered if, for example, we somehow miss the actual downtime start event. In this case, it should not trigger any notifications as it is clearly handled by the user. Yes, I know we don't have access to the `is_flapping` information with every state change event, but it's still better than nothing.